File: scripts/auto_complete.py
import json
import time


t = time.perf_counter()
with open("./result-korean-character.json", "r", encoding="utf-8") as f:
    data: dict[str, str] = json.load(f)


def auto_complete(word: str) -> list[str]:
    return [key for key in data.keys() if word in key]

print(auto_complete("chino"))
print(time.perf_counter() - t)

File: scripts/auto_complete.js
const fs = require('fs')

let start = new Date();

const jsonFile = fs.readFileSync('./result-korean-character.json', 'utf8')

const jsonData = JSON.parse(jsonFile);

const result = []

Object.keys(jsonData).forEach(key => {
    if (key.includes('chino')){
        result.push(key)
    }
})


console.log(result)
console.log((new Date() - start) / 1000)